crates/sparrow-runtime/src/metadata/prepared_metadata.rs

```diff
@@ -67,6 +70,39 @@ fn get_time_statistics(
 }
 
 impl PreparedMetadata {
+    pub fn try_from_data(
+        data_path: String,
+        data: &RecordBatch,
+        metadata_path: String,
+    ) -> anyhow::Result<Self> {
+        let prepared_schema = data.schema();
+
+        anyhow::ensure!(
+            prepared_schema.field(0).name() == "_time",
+            "First column of prepared files must be '_time'"
+        );
+
+        // Compute the time statistics directly from the data.
+        //
+        // TODO: We could instead just get this from the parquet metadata.
+        let time = data.column(0);
+        let time: &TimestampNanosecondArray = downcast_primitive_array(time.as_ref())?;
+
+        let num_rows = data.num_rows() as i64;
+        anyhow::ensure!(num_rows > 0, "Data should be non-empty");
+        let min_time = *time.values().iter().min().expect("non-empty");
+        let max_time = *time.values().iter().max().expect("non-empty");
```

Comment:
Is it cheap to calculate the min and max from the iterators? Is it reading from metadata or is it iterating through all the time values?

Reply:
It is iterating through the data -- but that should be cheaper than writing to Parquet to get the stats. Actually, since we've sorted we should be able to just use the first and last, which should be basically free.

And as noted in the other response -- we could do this from the Parquet footer stats more directly without having our own proto, which might be even simpler.

Reply:
Updated to use first/last.